zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithTerminationInstructionTest.java

```diff
@@ -316,15 +312,19 @@ record ->
                         || record.getValue().getElementId().equals(multiInstanceTaskId));
 
     assertThat(result)
-        .extracting(Record::getIntent, record -> record.getValue().getElementId())
+        .extracting(Record::getIntent, r -> r.getValue().getBpmnElementType())
         .containsSubsequence(
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, multiInstanceTaskId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATING, processId),
-            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATED, processId));
+            Tuple.tuple(
+                ProcessInstanceIntent.ELEMENT_ACTIVATED, BpmnElementType.MULTI_INSTANCE_BODY),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_ACTIVATED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, BpmnElementType.SCRIPT_TASK),
+            Tuple.tuple(
+                ProcessInstanceIntent.ELEMENT_COMPLETED, BpmnElementType.MULTI_INSTANCE_BODY),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATING, BpmnElementType.PROCESS),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATED, BpmnElementType.PROCESS));
```

Comment:
💭 This change could've been a separate commit with it's own message explaining why.